configure.ac

```diff
@@ -4558,6 +4558,8 @@ if test "$with_mimalloc" != no; then
   with_mimalloc=yes
   AC_DEFINE([WITH_MIMALLOC], [1], [Define if you want to compile in mimalloc memory allocator.])
   AC_SUBST([MIMALLOC_HEADERS], ['$(MIMALLOC_HEADERS)'])
+elif test "$disable_gil" = "yes"; then
```

Comment:
@colesbury 
Why we don't treat `--disable-gil` as the implicit option of `--with-mimalloc`.
IMO, If people enabled `--disable-gil` then `--with-malloc` should be enabled automatically as the seamless option.

Reply:
Plus, How about unifying `Py_GIL_DISABLED` and `WITH_MIMALLOC ` into just using `Py_GIL_DISABLED` until free-threaded build can choose the other malloc option (I am not sure at this moment).

Reply:
`--with-mimalloc` is enabled by default. You don't need to pass `--with-mimalloc`. You only run into the error if:

1) You configure with `--disable-gil --without-mimalloc`
2) You configure with `--disable-gil` and the compiler doesn't support `<stdatomic.h>` (e.g., GCC older than 4.8)

There was one place that checked both `Py_GIL_DISABLED` and `WITH_MIMALLOC`. I combined the checks.

Reply:
> --with-mimalloc is enabled by default

Ah, I missed this point, Thanks for letting me know :)